Mathlib/LinearAlgebra/InvariantBasisNumber.lean

```diff
@@ -11,34 +12,65 @@ import Mathlib.RingTheory.PrincipalIdealDomain
 /-!
 # Invariant basis number property
 
-We say that a ring `R` satisfies the invariant basis number property if there is a well-defined
-notion of the rank of a finitely generated free (left) `R`-module. Since a finitely generated free
-module with a basis consisting of `n` elements is linearly equivalent to `Fin n → R`, it is
-sufficient that `(Fin n → R) ≃ₗ[R] (Fin m → R)` implies `n = m`.
+## Main definitions
 
-It is also useful to consider two stronger conditions, namely the rank condition,
-that a surjective linear map `(Fin n → R) →ₗ[R] (Fin m → R)` implies `m ≤ n` and
-the strong rank condition, that an injective linear map `(Fin n → R) →ₗ[R] (Fin m → R)`
-implies `n ≤ m`.
+- `InvariantBasisNumber R` is a type class stating that `(Fin n → R) ≃ₗ[R] (Fin m → R)`
```

Comment:
Maybe make it clear that R is a (non-commutative) ring here? I certainly see "R" and assume "commutative ring" :-/

Reply:
Done.